### PR TITLE
bitstamp - fetchTransactionFees

### PR DESCRIPTION
**Pull request by AlgoQ**

### CODE REVIEW

File: js/bitstamp.js [hunk 1]
```diff
@@ -1140,36 +1140,53 @@ module.exports = class bitstamp extends Exchange {
...
+            if (id.indexOf ('_withdrawal_fee') >= 0) {
+                const currencyId = id.split ('_')[0];
+                const code = this.safeCurrencyCode (currencyId);
+                if (codes !== undefined && !this.inArray (code, codes)) {
+                    continue;
+                }
+                result[code] = {
+                    'deposit': undefined,
+                    'withdraw': this.safeNumber (response, id),
+                    'info': this.safeNumber (response, id),
```
Comment (carlosmiei) [inline js/bitstamp.js:1185]: @AlgoQ We're losing a lot of the original info here, what if, in this  `info` we added all related fields? 

For instance:
```
code[ZRX] = {
     (...),
      'info': {
            zrx_available: '0.00000000',
            zrx_balance: '0.00000000',
            zrx_reserved: '0.00000000',
            zrx_withdrawal_fee: '12.00000000',
      }
}
```
wdyt?

Comment (AlgoQ) [reply js/bitstamp.js:1185]: Only `zrx_available`, `zrx_balance`, `zrx_reserved` & `zrx_withdrawal_fee`?

Or also the other ones?
```
yfi_available: '0.00000000',
yfi_balance: '0.00000000',
yfi_reserved: '0.00000000',
yfi_withdrawal_fee: '0.00070000',
yfieur_fee: '0.000',
yfiusd_fee: '0.000',
```

Comment (carlosmiei) [reply js/bitstamp.js:1185]: @AlgoQ all of them that was just an example 🙂

File: js/bitstamp.js [hunk 2]
```diff
@@ -1140,36 +1140,75 @@ module.exports = class bitstamp extends Exchange {
...
+            }
+            if (id.indexOf ('_withdrawal_fee') >= 0) {
+                result[code]['withdraw'] = this.safeNumber (response, id);
+            }
+            prevCode = this.safeCurrencyCode (mainCurrencyId);
+            if (i === ids.length - 1 && infoObjectLen > 0) {
+                result[prevCode]['info'] = infoObject;
+            }
+        }
+        return result;
```
Comment (carlosmiei) [inline js/bitstamp.js:1211]: @AlgoQ something is off here: 
Javascript:
image

Python:
image

File: js/bitstamp.js [hunk 3]
```diff
@@ -1140,36 +1140,73 @@ module.exports = class bitstamp extends Exchange {
...
+                }
+            }
+            if (currencyId === mainCurrencyId) {
+                infoObject[id] = this.safeNumber (response, id);
+            }
+            if (id.indexOf ('_withdrawal_fee') >= 0) {
+                result[code]['withdraw'] = this.safeNumber (response, id);
+            }
+            prevCode = this.safeCurrencyCode (mainCurrencyId);
+            if ((i === ids.length - 1 && infoObjectLen > 0) || this.inArray (code, codes)) {
```
Comment (carlosmiei) [inline js/bitstamp.js:1205]: @AlgoQ this crashes when the user does not provide `codes` 
image

**Comment by carlosmiei** [comment]
LGTM ✅ 

```
 p bitstamp fetchTransactionFees '["USDT"]'
Python v3.10.6
CCXT v2.0.62
bitstamp.fetchTransactionFees(['USDT'])
{'USDT': {'deposit': None,
          'info': {'usdt_available': 0.61086,
                   'usdt_balance': 0.61086,
                   'usdt_reserved': 0.0,
                   'usdt_withdrawal_fee': 20.0},
          'withdraw': 20.0}}
```

**Review by carlosmiei — APPROVED** [review]

**Pull request merged**